src/main/java/frc/robot/leds/LedStrip.java

```diff
@@ -0,0 +1,181 @@
+package frc.robot.leds;
+
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.wpilibj.AddressableLED;
+import edu.wpi.first.wpilibj.AddressableLEDBuffer;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class LedStrip extends SubsystemBase {
+    private final AddressableLED ledStrip;
+    private AddressableLEDBuffer ledBuffer;
+    private int stripLength = 0;
+    private int offset = 0;
+    private int percentage = 100;
+
+    private double blinkTime = 1;
+    private double fadeDuration = 1;
+
+    private Color primary = new Color();
+    private Color secondary = new Color();
+    private Color currentColor = primary;
+    private Color fadeColor = primary;
+
+    private final Timer timer = new Timer();
+    private LedMode mode = LedMode.SOLID;
```

Comment:
I think that some of these variables are represented in the `LedState` class, so you can package them into a single object.